### PR TITLE
Clean up the public API, part 3

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 1]
```diff
@@ -375,7 +370,7 @@ object Types {
       * `<pre> . <sym>` is an actual argument reference, i.e., `pre` is not the
       * ThisType of `sym`'s owner, or a reference to `sym`'s owner.'
       */
-    def isArgPrefixOf(sym: Symbol)(using Context): Boolean =
+    private[tastyquery] def isArgPrefixOf(sym: Symbol)(using Context): Boolean =
```
Comment (bishabosha) [inline tasty-query/shared/src/main/scala/tastyquery/Types.scala:373]:
```suggestion
    private[tastyquery] final def isArgPrefixOf(sym: Symbol)(using Context): Boolean =
```

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 2]
```diff
@@ -416,25 +411,25 @@ object Types {
...
     /** The lower bound of a TypeBounds type, the type itself otherwise */
-    def lowerBound: Type = this match {
+    private[tastyquery] def lowerBound: Type = this match {
       case WildcardTypeBounds(bounds) => bounds.low
       case _                          => this
     }
 
     /** The upper bound of a TypeBounds type, the type itself otherwise */
-    def upperBound: Type = this match {
+    private[tastyquery] def upperBound: Type = this match {
```
Comment (bishabosha) [inline tasty-query/shared/src/main/scala/tastyquery/Types.scala:420]:
```suggestion
    private[tastyquery] final def upperBound: Type = this match {
```

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 3]
```diff
@@ -416,25 +411,25 @@ object Types {
...
       case WildcardTypeBounds(bounds) => bounds.high
       case _                          => this
     }
 
     /** Is self type bounded by a type lambda or AnyKind? */
-    def isLambdaSub(using Context): Boolean = false // TODO hkResult.exists
+    private[tastyquery] def isLambdaSub(using Context): Boolean = false // TODO hkResult.exists
```
Comment (bishabosha) [inline tasty-query/shared/src/main/scala/tastyquery/Types.scala:426]:
```suggestion
    private[tastyquery] final def isLambdaSub(using Context): Boolean = false // TODO hkResult.exists
```

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 4]
```diff
@@ -640,7 +630,7 @@ object Types {
       * Otherwise, a typebounds argument is dropped and the original type parameter
       * reference is returned.
       */
-    def argForParam(pre: Type, widenAbstract: Boolean = false)(using Context): Type = {
+    private[tastyquery] def argForParam(pre: Type, widenAbstract: Boolean = false)(using Context): Type = {
```
Comment (bishabosha) [inline tasty-query/shared/src/main/scala/tastyquery/Types.scala:633]:
```suggestion
    private[tastyquery] final def argForParam(pre: Type, widenAbstract: Boolean = false)(using Context): Type = {
```

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 5]
```diff
@@ -683,7 +673,7 @@ object Types {
       *
       *    T#A --> B    if A is bound to an alias `= B` in T
       */
-    def derivedSelect(prefix: Type)(using Context): Type =
+    private[tastyquery] def derivedSelect(prefix: Type)(using Context): Type =
```
Comment (bishabosha) [inline tasty-query/shared/src/main/scala/tastyquery/Types.scala:676]:
```suggestion
    private[tastyquery] final def derivedSelect(prefix: Type)(using Context): Type =
```

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 6]
```diff
@@ -926,23 +897,23 @@ object Types {
       * Applications of higher-kinded type constructors to wildcard arguments
       * are equivalent to existential types, which are not supported.
       */
-    def isUnreducibleWild(using Context): Boolean =
+    private[tastyquery] def isUnreducibleWild(using Context): Boolean =
```
Comment (bishabosha) [inline tasty-query/shared/src/main/scala/tastyquery/Types.scala:900]:
```suggestion
    private[tastyquery] final def isUnreducibleWild(using Context): Boolean =
```

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 7]
```diff
@@ -926,23 +897,23 @@ object Types {
...
+    private[tastyquery] override def findMember(name: Name, pre: Type)(using Context): Symbol =
       tycon match
         case tycon: TypeRef =>
           if tycon.symbol.isClass then tycon.findMember(name, pre)
           else ???
         case _ =>
           ???
 
-    def derivedAppliedType(tycon: Type, args: List[Type])(using Context): AppliedType =
+    private[tastyquery] def derivedAppliedType(tycon: Type, args: List[Type])(using Context): AppliedType =
```
Comment (bishabosha) [inline tasty-query/shared/src/main/scala/tastyquery/Types.scala:912]:
```suggestion
    private[tastyquery] final def derivedAppliedType(tycon: Type, args: List[Type])(using Context): AppliedType =
```

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 8]
```diff
@@ -926,23 +897,23 @@ object Types {
...
       if ((tycon eq this.tycon) && (args eq this.args)) this
       else AppliedType(tycon, args)
 
-    def map(op: Type => Type)(using Context): AppliedType =
+    private[tastyquery] def map(op: Type => Type)(using Context): AppliedType =
```
Comment (bishabosha) [inline tasty-query/shared/src/main/scala/tastyquery/Types.scala:916]:
```suggestion
    private[tastyquery] final def map(op: Type => Type)(using Context): AppliedType =
```

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 9]
```diff
@@ -952,7 +923,7 @@ object Types {
 
     override def underlying(using Context): Type = resType
 
-    def derivedExprType(resType: Type)(using Context): ExprType =
+    private[tastyquery] def derivedExprType(resType: Type)(using Context): ExprType =
```
Comment (bishabosha) [inline tasty-query/shared/src/main/scala/tastyquery/Types.scala:926]:
```suggestion
    private[tastyquery] final def derivedExprType(resType: Type)(using Context): ExprType =
```

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 10]
```diff
@@ -986,16 +955,16 @@ object Types {
...
-      resType: Type = this.resType
+      resType: Type = this.resultType
     )(using Context): This =
-      if (paramNames eq this.paramNames) && (paramInfos eq this.paramInfos) && (resType eq this.resType) then
+      if (paramNames eq this.paramNames) && (paramInfos eq this.paramInfos) && (resType eq this.resultType) then
         this.asInstanceOf[This]
       else newLikeThis(paramNames, paramInfos, resType)
 
-    def newLikeThis(paramNames: List[ThisName], paramInfos: List[PInfo], resType: Type)(using Context): This =
+    protected def newLikeThis(paramNames: List[ThisName], paramInfos: List[PInfo], resType: Type)(using Context): This =
```
[the remark this answers is not in the document]
Comment (bishabosha) [reply tasty-query/shared/src/main/scala/tastyquery/Types.scala:967]: or make it final

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 11]
```diff
@@ -1293,7 +1259,7 @@ object Types {
   case class AnnotatedType(typ: Type, annotation: Tree) extends TypeProxy with ValueType {
     override def underlying(using Context): Type = typ
 
-    def derivedAnnotatedType(typ: Type, annotation: Tree)(using Context): AnnotatedType =
+    private[tastyquery] def derivedAnnotatedType(typ: Type, annotation: Tree)(using Context): AnnotatedType =
```
Comment (bishabosha) [inline tasty-query/shared/src/main/scala/tastyquery/Types.scala:1262]:
```suggestion
    private[tastyquery] final def derivedAnnotatedType(typ: Type, annotation: Tree)(using Context): AnnotatedType =
```

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 12]
```diff
@@ -1310,7 +1276,9 @@ object Types {
       with ValueType {
     override def underlying(using Context): Type = parent
 
-    def derivedRefinedType(parent: Type, refinedName: Name, refinedInfo: TypeBounds)(using Context): Type =
+    private[tastyquery] def derivedRefinedType(parent: Type, refinedName: Name, refinedInfo: TypeBounds)(
```
Comment (bishabosha) [inline tasty-query/shared/src/main/scala/tastyquery/Types.scala:1279]:
```suggestion
    private[tastyquery] final def derivedRefinedType(parent: Type, refinedName: Name, refinedInfo: TypeBounds)(
```

**Review by bishabosha — CHANGES_REQUESTED** [review]
mostly comments about final modifier

**Comment by sjrd** [comment]
Hopefully I didn't miss any or your recommended `final`s. ^^

**Review by bishabosha — APPROVED** [review]

**Pull request merged**